### ISSUE TITLE
[Feature]: codeformer_fidelity (done) and codeformer_scale support

### ISSUE THREAD

**Issue by Arcitec** [Gourieff/comfyui-reactor-node#129]
#### codeformer_fidelity

After seeing what https://github.com/mav-rik/facerestore_cf can do with its `codeformer_fidelity` parameter, I realize that it's an incredibly important aspect of face restoration.

- GFPGan: Doesn't support it/doesn't use that parameter.
- CodeFormer: Requires that parameter. At 0, the face looks nothing like the input but becomes very aesthetic. At 0.5 it's a mix. At 1.0 it very closely matches the input face but becomes high resolution. This usually produces the best faces.

Without that parameter, face restore in ReActor node is almost useless.

I could set ReActor to face restore "none" and route the output to the separate "facerestore_cf" node, yes, but this is slower because that means it has to do face-detection twice. So it's better if `codeformer_fidelity` parameter is supported in ReActor.

It's gives a HUGE quality increase! :)

#### codeformer_scale

This 2nd parameter is something I just discovered now and haven't seen before. You can read about it in this article:

https://blog.segmind.com/best-settings-for-codeformer/

Basically, scale is how detailed and what resolution the output face will be. It's another huge quality increase. Look:

<img width="897" alt="image" src="https://github.com/Gourieff/comfyui-reactor-node/assets/38923130/8855b04c-1078-487b-9be4-811c3c5f0d4b">

#### Why?

Together, fidelity and scale makes codeformer face restore quality massively improved!

Would be a huge improvement for ReActor node. The current face restoration in reactor-node is operating at like 30% of the quality that can be achieved if those parameters are added. :)

**Comment by Gourieff** [comment]
Thanks for the info 👍 I will take a look

**Comment by ardha27** [comment]
> ### Feature description
> #### codeformer_fidelity
> After seeing what https://github.com/mav-rik/facerestore_cf can do with its `codeformer_fidelity` parameter, I realize that it's an incredibly important aspect of face restoration.
> 
> * GFPGan: Doesn't support it/doesn't use that parameter.
> * CodeFormer: Requires that parameter. At 0, the face looks nothing like the input but becomes very aesthetic. At 0.5 it's a mix. At 1.0 it very closely matches the input face but becomes high resolution. This usually produces the best faces.
> 
> Without that parameter, face restore in ReActor node is almost useless.
> 
> I could set ReActor to face restore "none" and route the output to the separate "facerestore_cf" node, yes, but this is slower because that means it has to do face-detection twice. So it's better if `codeformer_fidelity` parameter is supported in ReActor.
> 
> It's gives a HUGE quality increase! :)
> 
> #### codeformer_scale
> This 2nd parameter is something I just discovered now and haven't seen before. You can read about it in this article:
> 
> https://blog.segmind.com/best-settings-for-codeformer/
> 
> Basically, scale is how detailed and what resolution the output face will be. It's another huge quality increase. Look:
> 
> <img alt="image" width="897" src="https://private-user-images.githubusercontent.com/38923130/289235227-8855b04c-1078-487b-9be4-811c3c5f0d4b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI0NTc2MjcsIm5iZiI6MTcwMjQ1NzMyNywicGF0aCI6Ii8zODkyMzEzMC8yODkyMzUyMjctODg1NWIwNGMtMTA3OC00ODdiLTliZTQtODExYzNjNWYwZDRiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjEzVDA4NDg0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg1OWQyZjRjOGQ3Y2UxNjE5ZDM0ZjlhMDJmZmNkZDBlNTEyYjUzODI2YTE0Y2JiYzRhNDA4MjIwZDhjNDA0MmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.2vGbOm22kFBscSlMGVTI5-aFQh1rhLeKEjjRJZmJifk">
> #### Why?
> Together, fidelity and scale makes codeformer face restore quality massively improved!
> 
> Would be a huge improvement for ReActor node. The current face restoration in reactor-node is operating at like 30% of the quality that can be achieved if those parameters are added. :)

can i find "codeformer_scale" on comfyui?

**Comment by quantumdotai** [comment]
Came to Github literally to request this since I couldn't find the settings I'm normally used to when working with CodeFormer.

Awesome to see it's being taken into consideration already. Looking forward to it

**Comment by Gourieff** [comment]
codeformer_fidelity is implemented https://github.com/Gourieff/comfyui-reactor-node/commit/ea536a43c8527341f1426d19ca438b4a0329d9e1

**Comment by KINGLIFER** [comment]
May I at least inquire how to implement codeformer_scale

**Comment by Gourieff** [comment]
But why do you need this option? You can scale with any model you want via Upscaler's Nodes or via Ultimate Upscaler

**Comment by KINGLIFER** [comment]
No this is the face because of the limitations. I made something that upscales a face but why have all those nodes when one can do it itself.

**Comment by Gourieff** [comment]
Because it's not the best way to upscale, there are a lot of models that can provide better quality

**Comment by KINGLIFER** [comment]
for the details of a faceswap? Can you be more specific?
